addons/account/populate/account_move.py

```diff
@@ -0,0 +1,214 @@
+
+            # Add a last line containing the balance.
+            # For invoices, etc., it will be on the receivable/payable account.
+            lines += [get_line(
+                account=random.choice(balance_account_ids),
+                balance=sum(l[2]['credit'] - l[2]['debit'] for l in lines),
+                label='balance',
+                exclude_from_invoice_tab=move_type in self.get_invoice_types(include_receipts=True),
+            )]
+
+            return lines
+
+        def get_journal(random, values, **kwargs):
+            """Get a random journal depending on the company and the move_type.
+
+            :param random: seeded random number generator.
+            :param values (dict): the values already selected for the record.
+            :return (int): the id of the journal randomly selected
+            """
+            move_type = values['move_type']
+            company_id = values['company_id']
+            if move_type in self.get_sale_types(include_receipts=True):
+                journal_type = 'sale'
+            elif move_type in self.get_purchase_types(include_receipts=True):
+                journal_type = 'purchase'
+            else:
+                journal_type = 'general'
+            journal = search_journal_ids(company_id, journal_type)
+            return random.choice(journal)
+
+        def get_partner(random, values, **kwargs):
+            """Get a random partner depending on the company and the move_type.
+
+            The first 3/5 of the available partners are used as customer
+            The last 3/5 of the available partners are used as suppliers
+            It means 1/5 is both customer/supplier
+            -> Same proportions as in account.payment
+            :param random: seeded random number generator.
+            :param values (dict): the values already selected for the record.
+            :return (int, bool): the id of the partner randomly selected if it is an invoice document
+                                 False if it is a Journal Entry.
+            """
+            move_type = values['move_type']
+            company_id = values['company_id']
+            partner_ids = search_partner_ids(company_id)
+
+            if move_type in self.get_sale_types(include_receipts=True):
+                return random.choice(partner_ids[:math.ceil(len(partner_ids)/5*2)])
+            if move_type in self.get_purchase_types(include_receipts=True):
+                return random.choice(partner_ids[math.floor(len(partner_ids)/5*2):])
+            return False
+
+        company_ids = self.env['res.company'].search([('chart_template_id', '!=', False), ('id', 'in', self.env.registry.populated_models['res.company'])])
+
+        return [
+            ('move_type', populate.randomize(['entry', 'in_invoice', 'out_invoice', 'in_refund', 'out_refund', 'in_receipt', 'out_receipt'], [0.2, 0.3, 0.3, 0.07, 0.07, 0.03, 0.03])),
+            ('company_id', populate.randomize(company_ids.ids)),
```

Comment:
The company is related of the journal so this line is useless

Reply:
While this is true, it is easier to do it the way I did it because when I search for partners, I need to know the company too.
If I only set the journal, I would have to browse the journal to read its company.
Same for searching on accounts from the company.

If I set the company then search for its journals, the number of reads on the DB is small because the result is cached by company. If I did it the other way around, it would be cached by journal.